unblob/handlers/archive/xiaomi/hdr.py

```diff
@@ -106,7 +106,7 @@ def extract(self, inpath: Path, outdir: Path):
 
     def parse(self, file: File) -> Iterable[Tuple[Path, Chunk]]:
         header = self._struct_parser.parse(self.header_struct, file, Endian.LITTLE)
-        for offset in header.blob_offsets:
+        for offset in cast(Iterable, header.blob_offsets):
```

Comment:
I don't get these errors. For me pyright tells, it's `Any`. It is interesting, that you got `Never`. This means, that pyright thinks, that the method is diverting, i.e always raises an exception.

Reply:
There were some changes for `StructParser.parse()`: previously its return type was unspecified, and was interpreted as `Any`, it has been changed to `dissect.struct.Instance`.

The issue with `Instance` is, that it is a rather generic class. All the properties on it is generated according to the spec `parse()` receives. So `blob_offset` here can be found only inside of a struct spec string, and not in any Python code, so pyright have no chance knowing about it.

Reply:
These error messages have been copied out from a failed github action output: https://github.com/onekey-sec/unblob/actions/runs/4940612918/jobs/8832518378